docs/002-database.adoc

```diff
@@ -0,0 +1,37 @@
+= ADR 002 - Database
+Richard Holzeis <richard@10101.finance>
+3.0, July 29, 2022: AsciiDoc article template
+:toc:
+:icons: font
+:attributes: 2023-01-31
+
+This document outlines the decision how data is managed within 10101.
+
+== Decision
+
+* Database: Mobile app, coordinator and orderbook use different databases best suited for their corresponding platform. e.g. while the mobile phone may use Sqlite the coordinator is free to use any unmanaged or managed database that is best fit to deal with a higher user load.
+However, the chosen databases need to be open source and free to use.
+* Schema: 10101 will use a relational database for the mobile app as well as for the coordinator and order book. It is recommended to use align the schemas as best as possible to keep maintenance efforts as low as possible.
+However, the development of either schema must not be blocked by either platforms constraints. Dedicated adaptors for the different databases will have to be implemented and separately maintained.
+
+== Context
+
+Both mobile app and coordinator need to persist data. This data is mostly the same kind of data and can be categorized in the following.
+
+- *Domain Data*: Trading data like Orders, Trades, Positions, etc.
+- *Reference Data*: Trading Pair, Currency, etc.
+- *Technical Data*: DLC and Lightning data.
+- *Confidential Data*: Private or secret keys.
```

Comment:
This might be also covered in https://github.com/get10101/10101/issues/84

Reply:
yes exactly.. we might even think about using different databases for different purposes. e.g. i think for the technical data a nosql database might be better suited than a relational database.